gravitee-am-service/src/main/java/io/gravitee/am/service/impl/PasswordServiceImpl.java

```diff
@@ -72,4 +76,26 @@ public void validate(String password, PasswordSettings passwordSettings, User us
             });
         }
     }
+
+    /**
+     * Check the user password status
+     * @param user Authenticated user
+     * @return True if the password has expired or False if not
+     */
+    public boolean checkAccountPasswordExpiry(User user, Client client, Domain domain) {
+        Optional<PasswordSettings> passwordSettings = PasswordSettings.getInstance(client, domain);
+
+        /** If the expiryDate is null or set to 0 so it's disabled */
+        if (passwordSettings.isEmpty() ||
+                (passwordSettings.get().getExpiryDuration() == null
+                                || (passwordSettings.get().getExpiryDuration() != null && passwordSettings.get().getExpiryDuration() <= 0))) {
```

Comment:
No need to test `passwordSettings.get().getExpiryDuration() != null` here as it is already tested just before.
If it is null, the test on the `value <= 0` will not be executed